tests/instruments/keysight/test_keysightE3642A.py

```diff
@@ -0,0 +1,99 @@
+import pytest
+
+from pymeasure.test import expected_protocol
+from pymeasure.instruments.keysight.keysightE3642A import KeysightE3642A
+
+
+def test_init():
+    with expected_protocol(
+            KeysightE3642A,
+            [],
+    ):
+        pass  # Verify the expected communication.
+
+
+def test_current_limit_setter():
+    with expected_protocol(
+            KeysightE3642A,
+            [(b'CURR 2.670000', None)],
+    ) as inst:
+        inst.current_limit = 2.67
+
+
+@pytest.mark.parametrize("comm_pairs, value", (
+    ([(b'CURR?', None),
+      (b'++read eoi', b'+2.67000000E+00\n')],
```

Comment:
You're using a Prologix Adapter, and you used the test generator, aren't you?

The prologix adapter writes the `read eoi` message, which is not part of the instrument code itself.
You have to replace it by `None`.
As the previous message pair has already None as read message, you can combine them:
```suggestion
    ([(b'CURR?', b'+2.67000000E+00\n')],
```

Unfortunately, you have to do that manually afterwards, due to the Prologix Adapter.

Reply:
Thanks for letting me know!